Test pycffi fails on mac as print returns warnings in addition to path

`pytest spy/tests/compiler/test_py_cffi.py`

```shell
______________________________________________________ TestPyCFFI.test_simple[py-cffi] ______________________________________________________

self = <spy.tests.compiler.test_py_cffi.TestPyCFFI object at 0x1069356d0>

    def test_simple(self):
>       mod = self.compile(
        """
        def add(x: i32, y: i32) -> i32:
            return x + y
        """)

spy/tests/compiler/test_py_cffi.py:8:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
spy/tests/support.py:217: in compile
    outfile = backend.build() # e.g. 'test.wasm' or 'test.mjs'
spy/backend/c/cbackend.py:119: in build
    return cffi_build(self.build_script)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

build_script = local('/private/var/folders/wt/y_7lt20d2239r1jcfxq8qrx40000gn/T/pytest-of-tedpatrick/pytest-7/test_simple_py_cffi_0/build/cffi/_test-cffi-build.py')

    def cffi_build(build_script: py.path.local) -> py.path.local:
        """
        Generate a CPython extension module by running the cffi-build.py
        script produced by spy.backend.c.cffiwriter.
        """
        cmdline = [sys.executable, str(build_script)]
        d = build_script.dirpath()
        with chdir(d):
            proc = robust_run(cmdline)
        out = proc.stdout.decode('utf-8')
        sofile = py.path.local(out.strip())
>       assert sofile.exists()
E       AssertionError

spy/build/cffi.py:30: AssertionError
========================================================== short test summary info ==========================================================
FAILED spy/tests/compiler/test_py_cffi.py::TestPyCFFI::test_simple[py-cffi] - AssertionError
```


PDB Output of print(sofile)
```
(Pdb) print(sofile)
/Users/tedpatrick/Dev/spy/ld: warning: duplicate -rpath '/opt/anaconda3/lib' ignored
/private/var/folders/wt/y_7lt20d2239r1jcfxq8qrx40000gn/T/pytest-of-tedpatrick/pytest-8/test_simple_py_cffi_0/build/cffi/_test.cpython-312-darwin.so
```

--- a/spy/build/cffi.py
+++ b/spy/build/cffi.py
@@ -5,8 +5,9 @@ import py.path
 import contextlib
 from spy.util import robust_run
 
+
 @contextlib.contextmanager
-def chdir(path: str|py.path.local) -> Any:
+def chdir(path: str | py.path.local) -> Any:
     """Context manager to temporarily change directory."""
     old_dir = os.getcwd()
     try:
@@ -25,7 +26,8 @@ def cffi_build(build_script: py.path.local) -> py.path.local:
     d = build_script.dirpath()
     with chdir(d):
         proc = robust_run(cmdline)
-    out = proc.stdout.decode('utf-8')
+    # The generated .so file is expected to be in the stdout of the build script
+    out = proc.stdout.decode("utf-8")
     sofile = py.path.local(out.strip())
     assert sofile.exists()
     return sofile

--- a/spy/util.py
+++ b/spy/util.py
@@ -127,11 +127,8 @@ def robust_run(
     """
     cmdline_s = [str(x) for x in cmdline]
     #print(" ".join(cmdline_s))
-    proc = subprocess.run(
-        cmdline_s,
-        stdout=subprocess.PIPE,
-        stderr=subprocess.STDOUT
-    )
+    # Use capture_output=True to capture stdout and stderr separately
+    proc = subprocess.run(cmdline_s, capture_output=True)
     if proc.returncode != 0:
         FORCE_COLORS = True
         lines = ["subprocess failed:"]

